libstuff/SData.h

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+#include "libstuff.h"
+
+// --------------------------------------------------------------------------
+// A very simple HTTP-like structure consisting of a method line, a table,
+// and a content body.
+// --------------------------------------------------------------------------
+struct SData {
+    // Public attributes
+    string methodLine;
+    STable nameValueMap;
+    string content;
+
+    // Constructors
+    SData();
+
+    // Initializes a new SData from a string.  If the string provided is not
+    // an entire HTTPs like message, the string is used as the methodLine.
+    SData(const string& fromString);
+
+    // Allow forwarding emplacements directly so SData can act like `std::map`.
+    template <typename... Ts>
+    pair<decltype(nameValueMap)::iterator, bool> emplace(Ts&&... args) {
+        return nameValueMap.emplace(forward<Ts>(args)...);
+    }
+
+
+    // Operators
+    // This version creates an entry, if necessary, and returns a reference
+    string& operator[](const string& name);
```

Comment:
I bet we don't even want this, honestly. What if we remove it entirely? Does anything break?

Reply:
as discussed in slack, this breaks inserting into an SData object

Reply:
I was only considering the case of looking up/inserting blank values. Turns out we also want non-blank values.